### ISSUE TITLE
Adding default arg to Field overwrites value that already exists.

### ISSUE THREAD

**Issue by jerber** [octabytes/FireO#43]
It appears that when you add a default to a field, even if that field exists in the DB, it will be overwritten by the default value.

For example:

class BillingHistory(Model):
	id = IDField()
	amount = NumberField(required=True, default=15)

b = BillingHistory(id = '123', amount = 11)
b.save() # amount == 11 in Firestore
b.update() # amount == 15 in Firestore (but should have stayed to 11 since it existed)

**Comment by AxeemHaider** [comment]
This is a bug and never notice before thanks for pointing to this. Will fix soon.
But there is no need to call `update()` method without assigning any new value.
Why you are calling `update()` method after `save()` without assigning any new value ?

**Comment by jerber** [comment]
It also happens when you assign a different value in the update. For example:

b = BillingHistory.collection.get('BillingHistory/123') # amount == 11
b.lastPaymentId = '1082902'
b.update() # amount == 15 (but should have stayed to 11)

**Comment by AxeemHaider** [comment]
It is because you are calling `update()` after `save()` without changing anything. 

This work fine if you remove the first `update()` method which you are calling after `save()`
```python
b = BillingHistory.collection.get('BillingHistory/123') # amount == 11
b.lastPaymentId = '1082902'
b.update() # amount == 15 (but should have stayed to 11)
````

**Working Example**

This code work fine for me.
```python
class BillingHistory(Model):
     amount = NumberField(required=True, default=15)
     lastPaymentId = NumberField()

b = BillingHistory()
b.amount = 11
b.save()

# updating value
b = BillingHistory.collection.get(b.key)
b.lastPaymentId = 123
b.update()

# printing value
print(b.to_dict())
# amount = 11
# lastPaymentId = 123
```

### Tips:
1. Don't generate `key` yourself like **BillingHistory/id-123** instead of this use build in attribute `key` like `b.key`. BillingHistory object contain the `key` attribute which hold the `key` for this object. 

2.  I see `b.lastPaymentId` it seems your are pointing other document best way to do that is to use [ReferenceField](https://octabyte.io/FireO/fields/reference-field/)

**Comment by AxeemHaider** [comment]
issue is fixed in `version 1.2.3` now you can call `update` method and value will not be override.

```python
class BillingHistory(Model):
id = IDField()
amount = NumberField(required=True, default=15)

b = BillingHistory(id = '123', amount = 11)
b.save() # amount == 11 in Firestore
b.update() # amount == 11 in Firestore
```

To update `NestedModel` [See this](https://octabyte.io/FireO/querying-data/updating-data/#update-nestedmodel)
